server/src/main/java/org/apache/iotdb/db/qp/executor/PlanExecutor.java

```diff
@@ -1524,8 +1540,56 @@ public void insertTablet(InsertMultiTabletPlan insertMultiTabletPlan)
     }
   }
 
+  private void insertTabletParallel(InsertMultiTabletPlan insertMultiTabletPlan)
+      throws BatchProcessException {
+    updateInsertTabletsPool(insertMultiTabletPlan.getDifferentStorageGroupsCount());
+
+    List<InsertTabletPlan> planList = insertMultiTabletPlan.getInsertTabletPlanList();
+    List<Future<?>> futureList = new ArrayList<>();
+
+    Map<Integer, TSStatus> results = insertMultiTabletPlan.getResults();
+    for (InsertTabletPlan plan : planList) {
+      Future<?> f =
+          insertionPool.submit(
+              () -> {
+                insertTablet(plan);
+                return null;
+              });
+      futureList.add(f);
+    }
+    for (int i = 0; i < futureList.size(); i++) {
+      try {
+        futureList.get(i).get();
```

Comment:
```suggestion
        futureList.get(i).get();
```
Handle the exception here :D